### PR TITLE
Inject missing dialects in E2E passes

### PR DESCRIPTION
**Pull request by marbre**

### CODE REVIEW

**Comment by marbre** [comment]
This fixes all most E2E tests. However, `E2E/lower-shaped-results-to-memref.mlir` is failing if `-split-input-file` is passed to `npcomp-opt`:
```
LLVM ERROR: Building op `shape.get_extent` but it isn't registered in this MLIRContext
```
Hence, we still missing to register the shape dialect somewhere.

File: lib/E2E/TensorToMemref/LowerShapedResultsToMemref.cpp [hunk 1]
```diff
@@ -242,7 +242,11 @@ namespace {
 // TODO: Don't hardcode the lowering for every op in this one pass.
 class LowerShapedResultsToMemref
     : public LowerShapedResultsToMemrefBase<LowerShapedResultsToMemref> {
-  void runOnOperation() {
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<linalg::LinalgDialect, scf::SCFDialect>();
```
Comment (silvasean) [inline lib/E2E/TensorToMemref/LowerShapedResultsToMemref.cpp:246]: can you add shape::ShapeDialect here? I think that should fix the test.

**Review by silvasean — APPROVED** [review]
Thanks!

File: lib/E2E/TensorToMemref/LowerShapedResultsToMemref.cpp (hunk 1 above)
Comment (marbre) [reply lib/E2E/TensorToMemref/LowerShapedResultsToMemref.cpp]: Just noticed my mistake and fixed it.

**Pull request merged**